classes/announcement/AnnouncementDAO.inc.php

```diff
@@ -279,15 +279,16 @@ function getNumAnnouncementsByAssocId($assocType, $assocId, $numAnnouncements, $
 	 * @return object DAOResultFactory containing matching Announcements
 	 */
 	function getAnnouncementsNotExpiredByAssocId($assocType, $assocId, $rangeInfo = null) {
+		$currentDate = $this->getNextDay();
 		$result = $this->retrieveRange(
 			'SELECT *
 			FROM announcements
 			WHERE assoc_type = ?
 				AND assoc_id = ?
-				AND (date_expire IS NULL OR DATE(date_expire) > CURRENT_DATE)
```

Comment:
I think the ADODB `$sysDate` construct might be better to use?

Reply:
You mean generating the date with ADOdb?

Reply:
`$sysDate` in ADODB contains the name of the SQL function to use according to the DBMS. The date would still be generated on the SQL server.

Reply:
Replaced by https://github.com/pkp/pkp-lib/commit/f338bb47f613931cdf6defe3a7fd2e8c15540638.